src/test/java/seedu/address/logic/commands/BuyCommandTest.java

```diff
@@ -160,6 +185,37 @@ public boolean hasGood(Good good) {
         public void addGood(Good good) {
             inventory.add(good);
         }
+
+        @Override
+        public ObservableList<Supplier> getFilteredSupplierList() {
+            ObservableList<Supplier> supplierList = FXCollections.observableArrayList();
+            supplierList.add(supplierSellingBoughtGood);
+            return supplierList;
+        }
+    }
+
+    /**
+     * Class to stub the commit method of Model, and adds a certain person to the model when commit is called.
+     */
+    private class ModelStubCommit extends ModelStub {
+        private boolean isCommitted = false;
+        private ArrayList<Supplier> supplierList = new ArrayList<>();
+
+        public boolean isCommitted() {
+            return isCommitted;
+        }
+
+        @Override
+        public void commit() {
+            isCommitted = true;
+        }
+
+        @Override
+        public ObservableList<Supplier> getFilteredSupplierList() {
+            ObservableList<Supplier> supplierList = FXCollections.observableArrayList();
+            supplierList.add(supplierSellingBoughtGood);
+            return supplierList;
+        }
     }
```

Comment:
Why have this class here?

Reply:
Oops, left over from failed attempt to group related ModelStubs together. Removed.